backend/src/services/kms/kms-service.ts

```diff
@@ -167,35 +167,120 @@ export const kmsServiceFactory = ({
   };
 
   const getProjectSecretManagerKmsKeyId = async (projectId: string) => {
-    const keyId = await projectDAL.transaction(async (tx) => {
-      const project = await projectDAL.findById(projectId, tx);
-      if (!project) {
-        throw new BadRequestError({ message: "Project not found" });
+    let project = await projectDAL.findById(projectId);
+    if (!project) {
+      throw new BadRequestError({ message: "Project not found" });
+    }
+
+    if (!project.kmsSecretManagerKeyId) {
+      // create default kms key for certificate service
+      const lock = await keyStore
```

Comment:
We do this so that multiple requests to create the default KMS doesn't create conflict?

Reply:
yup exactly

Reply:
last thing we want is for data key to be encrypted by a stale KMS